packages/echo-core/src/__tests__/actions/coreActions/extensions.test.ts

```diff
@@ -0,0 +1,151 @@
+import { registerExtension, registerMultipleExtensions } from '../../../actions/coreActions/extensions';
+import { readState } from '../../../actions/coreActions/globalActions';
+import { getCoreContext } from '../../../state/globalState';
+import { ExtensionRegistration } from './../../../types/registry/extension.types';
+
+describe('extensions', () => {
+    describe('registerExtension()', () => {
+        it('should register and unregister a given extension into the state', () => {
+            // when
+            const unregisterExtension = registerExtension(extension);
+            let registeredExtensions = getExtensions();
+
+            // then
+            expect(registeredExtensions).toEqual({
+                MyFavEchopediaWebComponent: [
+                    { component: mockComponent, extends: 'MyFavEchopediaWebComponent', key: 'unique-key' }
+                ]
+            });
+
+            // when
+            unregisterExtension();
+            registeredExtensions = getExtensions();
+
+            // then
+            expect(registeredExtensions).toStrictEqual({
+                MyFavEchopediaWebComponent: []
+            });
+        });
+
+        it('should ignore the registration if the passed key already exists', () => {
+            // when
+            const unregisterExtension = registerExtension(extension);
+            registerExtension(extension);
```

Comment:
we should probably test unregistering twice? to see that it doesn't crash

Reply:
Good point, will add a test for that as well.